pkg/controllers/metrics/nodepool/controller.go

```diff
@@ -131,11 +146,23 @@ func getLimits(nodePool *v1.NodePool) corev1.ResourceList {
 	return corev1.ResourceList{}
 }
 
-func makeLabels(nodePool *v1.NodePool, resourceTypeName string) prometheus.Labels {
-	return map[string]string{
+func makeLabels(ctx context.Context, nodePool *v1.NodePool, resourceTypeName string) prometheus.Labels {
+	opts := options.FromContext(ctx)
+	labels := prometheus.Labels{
 		resourceTypeLabel: resourceTypeName,
 		nodePoolNameLabel: nodePool.Name,
 	}
+
+	// Add values for additional labels from CLI flags
+	for _, labelKey := range opts.AdditionalNodePoolMetricLabels {
+		if value, ok := nodePool.Labels[labelKey]; ok {
+			labels[labelKey] = value
+		} else {
+			labels[labelKey] = ""
+		}
```

Comment:
already produces empty string on missing
```suggestion
		labels[labelKey] = nodePool.Labels[labelKey]
```